src/test/java/io/supertokens/test/session/SessionGetJWTDataTest.java

```diff
@@ -0,0 +1,195 @@
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    // * - create session with some JWT payload -> verify to see payload is proper -> pass null to
+    // changeJWTPayloadInDatabase
+    // *  function -> check that JWT payload has not changed is reflected
+    @Test
+    public void testVerifyJWTPayloadSetPayloadToNullUsingSessionHandle() throws Exception {
+        String[] args = {"../"};
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        //createSession with JWT payload
+        String userId = "userId";
+        JsonObject userDataInJWT = new JsonObject();
+        userDataInJWT.addProperty("key", "value");
+        JsonObject userDataInDatabase = new JsonObject();
+        userDataInDatabase.addProperty("key", "value");
+
+
+        SessionInformationHolder sessionInfo = Session
+                .createNewSession(process.getProcess(), userId, userDataInJWT, userDataInDatabase, false);
+
+        // verify to see payload is proper
+        assertEquals(sessionInfo.session.userDataInJWT, userDataInJWT);
+
+        //change JWT payload to be null
+        Session.updateSession(process.getProcess(), sessionInfo.session.handle, userDataInDatabase, null, null);
+
+        //check that jwtData does not change
+        assertEquals(Session.getSession(process.getProcess(), sessionInfo.session.handle).userDataInJWT, userDataInJWT);
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    // * - create session -> let it expire, remove from db -> call get function -> make sure you get unauthorised error
+    @Test
+    public void testExpireSessionCallGetAndCheckUnauthorised() throws Exception {
+
+        String[] args = {"../"};
+
+        Utils.setValueInConfig("access_token_validity", "1");// 1 second validity
+        Utils.setValueInConfig("refresh_token_validity", "" + 1.0 / 60);// 1 second validity (value in mins)
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        //createSession with JWT payload
+        String userId = "userId";
+        JsonObject userDataInJWT = new JsonObject();
+        userDataInJWT.addProperty("key", "value");
+        JsonObject userDataInDatabase = new JsonObject();
+        userDataInDatabase.addProperty("key", "value");
+
+
+        SessionInformationHolder sessionInfo = Session
+                .createNewSession(process.getProcess(), userId, userDataInJWT, userDataInDatabase, false);
+
+        //let it expire, remove from db
+
+        Thread.sleep(2000);
+        String[] sessionHandles = {sessionInfo.session.handle};
+        Session.revokeSessionUsingSessionHandles(process.getProcess(), sessionHandles);
```

Comment:
Should you need to call revoke at all? If the refresh token has also expired, the session should be as good as revoed.. So maybe add another test that does the same thing but does not call revoke (this may require changes to your logic for fetching session info to check the expiry time)

Reply:
Yeah I did think of that but I saw it being done in another test so stuck with it. Adding a different test for this is a good idea, on it

Reply:
Done